presto-main/src/main/java/com/facebook/presto/operator/scalar/JsonOperators.java

```diff
@@ -341,6 +349,49 @@ public static Slice castFromBoolean(@SqlType(BOOLEAN) boolean value)
         }
     }
 
+    @ScalarOperator(CAST)
+    @TypeParameter("T")
+    @SqlType(JSON)
+    public static Slice castFromArray(@TypeParameter("T") Type elementType, @SqlType("array(T)") Block value)
+            throws IOException
+    {
+        ArrayType arrayType = new ArrayType(elementType);
+        checkCondition(canCastToJson(arrayType), INVALID_CAST_ARGUMENT, "Cannot cast JSON from %s", arrayType);
```

Comment:
This will be checked for every row. @haozhun : Anyway to avoid this? Can we use the `SqlOperator` interface and check this in `specialize`?